src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,230 @@
+ticket (TGT) can be used for single sign-on into other Kerberos-enabled
+services.
+
+For the purpose of this feature, passkey is a FIDO2 compatible device supported
+by the libfido2 library.
+
+Use cases
+---------
+As a user, not only do I want to walk to any system on a network and log in
+(desktop and/or terminal) using a passkey device connected to the system, but I
+also want to get a Kerberos ticket to identify myself to other services.
+
+Prerequisites
+-------------
+To register a new
+`pre-authentication data identifier <https://www.iana.org/assignments/kerberos-parameters/kerberos-parameters.xhtml#pre-authentication>`__
+in Kerberos for passkey it would be good to first finalize the IdP plugin
+design and security analysis requested by this community.
+
+Solution overview
+-----------------
+The objective is to locally authenticate and issue a Kerberos ticket to a
+centrally managed user. For that purpose SSSD will need to be modified to
+accommodate the new workflow, while still allowing the old one, where the user
+was authenticated locally without issuing any Kerberos ticket. The new workflow
+will be used for online authentication against an IPA server. The old workflow
+will be used for any offline authentication and for AD and other LDAP servers.
+
+SSSD provides a pre-authentication method implementation (pre-auth plugin) to
+MIT Kerberos library. When both the libkrb5 and the KDC support the same
+pre-authentication method, KDC will respond to an AS-REQ message with a request
+to provide an assertion data required by a pre-authentication method. SSSD's
+pre-auth plugin will obtain the assertion data from the passkey. This
+information will be sent by libkrb5 back to the KDC server as a part of the
+AS-REQ exchange. This in turn will relay the request to the ipa-otpd, which
+will retrieve the user information and ask the passkey_child to validate the
+assertion. Once validated the Kerberos KDC will issue a ticket and send it to
+SSSD via libkrb5.
+
+.. image:: ../images/passkey_kerberos_1.png
+  :alt: Architecture diagram
+
+Implementation details
+----------------------
+
+General overview
+****************
+When SSSD support for passkey authentication is enabled, libkrb5 would
+advertise a support for passkey during AS-REQ request. If information about the
+Kerberos principal on the KDC side allows use of passkey authentication, KDC
+would respond with a pre-authentication data to choose passkey authentication.
+
+The following sequence diagram illustrates the workflow taking into account all
+the components involved in it:
+
+.. image:: ../images/passkey_kerberos_2.png
+  :alt: Sequence diagram
+
+* The user will try to login with a centrally managed account in the system.
+
+* PAM will relay this request to SSSD’s PAM responder.
+
+* The PAM responder will retrieve the user information from the cache, which
+  will provide this information including the authentication protocol (PASSKEY).
```

Comment:
I'd like to simplify the sequence diagram to avoid having a very big diagram that nobody understands. That's why I assumed that the cache already contained the user attributes, including the passkey mapping data. Does that make sense?

Reply:
Understood and fixed. Please check the new version